diff --git a/upstream/vyos-build/scripts/package-build/linux-kernel/build-linux-firmware.sh b/upstream/vyos-build/scripts/package-build/linux-kernel/build-linux-firmware.sh
index 5c7e93f..7d5961f 100755
--- a/upstream/vyos-build/scripts/package-build/linux-kernel/build-linux-firmware.sh
+++ b/upstream/vyos-build/scripts/package-build/linux-kernel/build-linux-firmware.sh
@@ -21,7 +21,7 @@ if [ ! -d ${LINUX_FIRMWARE} ]; then
 fi
 
 # Retrieve firmware blobs from source files
-FW_FILES=$(find ${KERNEL_DIR}/debian/linux-image-${KERNEL_VERSION}${KERNEL_SUFFIX}/lib/modules/${KERNEL_VERSION}${KERNEL_SUFFIX}/kernel/drivers/net -name *.ko | xargs modinfo | grep "^firmware:" | awk '{print $2}')
+FW_FILES=$(find ${KERNEL_DIR}/debian/linux-image-${KERNEL_VERSION}${KERNEL_SUFFIX}/lib/modules/${KERNEL_VERSION}${KERNEL_SUFFIX}/kernel/drivers/net \( -name '*.ko' -o -name '*.ko.xz' \) | xargs modinfo | grep "^firmware:" | awk '{print $2}')
 
 # Debian package will use the descriptive Git commit as version
 GIT_COMMIT=$(cd ${CWD}/${LINUX_FIRMWARE}; git describe --always)
